src/main/java/net/minecraftforge/event/entity/player/PlayerInteractEvent.java

```diff
@@ -326,6 +327,42 @@ public LeftClickEmpty(EntityPlayer player, @Nonnull ItemStack stack)
     }
 
     /**
+     * This event is fired on the client side when a player middle clicks while aiming on a block or entity.
+     * There is no update packet for it, you have to send your own when needed.
+     * You can change the Result ItemStack here or set it to {@link ItemStack#EMPTY} to prevent further updating.
+     * Cancel the event if you want that nothing happens.
+     */
+    @Cancelable
+    public static class Pick extends PlayerInteractEvent
+    {
+        private RayTraceResult target;
+        private ItemStack currentResult;
+        public Pick(EntityPlayer player, RayTraceResult target, ItemStack currentResult){
+            super(player, EnumHand.MAIN_HAND, target.typeOfHit == RayTraceResult.Type.BLOCK ? target.getBlockPos() : new BlockPos(target.entityHit), target.typeOfHit == RayTraceResult.Type.BLOCK ? target.sideHit : null);
+            this.target = target;
+            this.currentResult = currentResult;
+        }
+
+        /**
+         * @return The same RayTraceResult as {@link net.minecraft.client.Minecraft#objectMouseOver}
+         */
+        public RayTraceResult getRayTraceResult(){
+            return target;
+        }
+
+        /**
+         * @return The current analyzed result for the RayTraceResult
```

Comment:
drop the word "analyzed" (or drop the javadoc entirely as it should be self explanatory)